animated-transformer/src/lib/gtensor/gtensor.spec.ts

```diff
@@ -896,4 +894,29 @@ describe('gtensor', () => {
     // const attendedValues2 = attentionHeadFn2(batchedInput); // Has error, yay, but what a mess...
     const attendedValues3 = attentionHeadFn2(oneInput);
   });
+
+  it('simple triangular matrix', async () => {
+    const g1 = new gtensor.GTensor(
+      tf.tensor([
+        [
+          [
+            [0, 1, 2],
+            [3, 4, 5],
+            [6, 7, 8],
+          ],
+        ],
+      ]),
+      ['heads', 'batch', 'Pos1', 'Pos2'],
+    );
+    const g1tril = gtensor.makeTriangularMatrix(g1.dim['Pos1'].size, ['Pos1', 'Pos2'], 0, 42);
```

Comment:
g1 doesn't do much, simpler for this test to just have the constant 3 here. 

Clean up comments below too?